functions/get_postgres_hosts.pp

```diff
@@ -0,0 +1,18 @@
+function rsan::get_postgres_hosts() {
+  if $settings::storeconfigs {
+    $postgres_hosts =
+                puppetdb_query('resources[certname] {
+                    type = "Class" and
+                    title = "pe_postgresql::server::install" and
+                    nodes {
+                      deactivated is null and
+                      expired is null
+                    }
+                  }').map |$data| { $data['certname'] }
+  } else {
+    $postgres_hosts = []
```

Comment:
$postgres_hosts would need to be returned in a format 
puppet_metrics_dashboard:: postgres_host_list expects

Reply:
So in this case, puppet_metrics_dashboard::postgres_host would need to be an array of elements, with each element being a postgres host. I've been looking at https://puppet.com/docs/puppet/7.0/lang_data_type.html

Initially, part of me thinks that I need to some thing like:

```
function rsan::get_postgres_hosts() {
    **$postgres_hosts => Array**
    $postgres_hosts =
                puppetdb_query('resources[certname] {
                    type = "Class" and
                    title = "Pe_postgresql::Server::Install" and
                    nodes {
                      deactivated is null and
                      expired is null
                    }
                  }').map |$data| { $data['certname'] }
  }
```

My thinking is that by setting the data type beforehand - puppetdb_query will fill in the elements to that array as appropriate? 


But I've also come across https://puppet.com/docs/puppet/7.0/lang_data_type.html#lang_data_type_usage

Where there is the 'Type' function. In this instance, I would do something like get the data type from the $postgres_hosts **after** the puppetdb_query is finished, then if the data type of $postgres_hosts =/ array, I set it as so?